src/libcore/iter.rs

```diff
@@ -2731,7 +2747,11 @@ pub trait ExactSizeIterator: Iterator {
     /// implementation, you can do so. See the [trait-level] docs for an
     /// example.
     ///
+    /// This function has same performance requirements and safety
```

Comment:
Changed to "This function has the same safety guarantees as `size_hint()` function".